extensions/amp-consent/0.1/test/test-consent-config.js

```diff
@@ -153,6 +155,134 @@ describes.realWin('ConsentConfig', {amp: 1}, env => {
+            'geoOverride': {
+              'nafta': {
+                'consentRequired': true,
+              },
+              'waldo': {
+                'checkConsentHref': 'https://example.com/check-consent',
+                'consentRequired': 'remote',
+              },
+              'geoGroupUnknown': {
+                'checkConsentHref': 'https://example.com/check-consent',
+                'consentRequired': true,
+              },
+            },
+          })
+        );
+      });
+
+      it('does not merge', () => {
+        sandbox.stub(Services, 'geoForDocOrNull').returns(
+          Promise.resolve({
+            isInCountryGroup() {
+              return false;
+            },
+          })
+        );
+
+        const consentConfig = new ConsentConfig(element);
+        expect(
+          consentConfig.getConsentConfigPromise()
+        ).to.eventually.deep.equal(
+          dict({
+            'consentInstanceId': 'abc',
+            'consentRequired': false,
+            'checkConsentHref': '/override',
+          })
+        );
+      });
+
+      it('Overrides single field', () => {
+        sandbox.stub(Services, 'geoForDocOrNull').returns(
+          Promise.resolve({
+            isInCountryGroup(geoGroup) {
+              if (geoGroup === 'nafta') {
+                return GEO_IN_GROUP.IN;
+              }
+              return GEO_IN_GROUP.NOT_IN;
+            },
+          })
+        );
+
+        const consentConfig = new ConsentConfig(element);
+        expect(
+          consentConfig.getConsentConfigPromise()
+        ).to.eventually.deep.equal(
+          dict({
+            'consentInstanceId': 'abc',
+            'consentRequired': true,
+            'checkConsentHref': '/override',
+          })
+        );
+      });
+
+      it('Overrides both fields', () => {
+        sandbox.stub(Services, 'geoForDocOrNull').returns(
+          Promise.resolve({
+            isInCountryGroup(geoGroup) {
+              if (geoGroup === 'waldo') {
+                return GEO_IN_GROUP.IN;
+              }
+              return GEO_IN_GROUP.NOT_IN;
+            },
+          })
+        );
+
+        const consentConfig = new ConsentConfig(element);
+        expect(
+          consentConfig.getConsentConfigPromise()
+        ).to.eventually.deep.equal(
+          dict({
+            'consentInstanceId': 'abc',
+            'checkConsentHref': 'https://example.com/check-consent',
+            'consentRequired': 'remote',
+          })
+        );
+      });
+
+      it('Overrides null fields', () => {
+        sandbox.stub(Services, 'geoForDocOrNull').returns(
+          Promise.resolve({
+            isInCountryGroup(geoGroup) {
+              if (geoGroup === 'geoGroupUnknown') {
+                return GEO_IN_GROUP.IN;
+              }
+              return GEO_IN_GROUP.NOT_IN;
+            },
+          })
+        );
+
+        const json = JSON.parse(element.textContent);
+        delete json.consentRequired;
```

Comment:
for readability, i would prefer hardcoding a minimum config that serve your verification purpose here instead of manipulating an existing test data used for other purpose.

you might want to have a helper method to build ConsentConfig with a given JSON